srcpkgs/ugdb/template

```diff
@@ -0,0 +1,21 @@
+# Template file for 'ugdb'
+pkgname=ugdb
+version=0.1.7
+revision=1
+build_style=cargo
+hostmakedepends="pkg-config libgit2-devel"
+makedepends="libgit2-devel"
+short_desc="Alternative TUI for gdb"
+maintainer="Érico Nogueira <ericonr@disroot.org>"
+license="MIT"
+homepage="https://github.com/ftilde/ugdb"
+distfiles="${homepage}/archive/${version}.tar.gz"
+checksum=b71e344ef385d00b54338fcb98e0add628ff16e4ee6c0c8761f8788bb56a99d1
+
+pre_build() {
+	export LIBGIT2_SYS_USE_PKG_CONFIG=yes
```

Comment:
how many times do i have to say not to use this thing

Reply:
It only seems to be used in `build.rs`, so I figured it would be ok. Otherwise it tries to build `libgit2` just to include commit information into the program, which doesn't even exist in the tarball.

Reply:
Okay, I think I managed to fix it with a sed replacement instead. It's similar to the one used in `git-absord`. However, I seem to need to add `libgit2-devel` to `makedepends` as well, because otherwise the build system passes flags like `-m64` to gcc for the linking step. Should I report this as an error somewhere? It certainly seems like one.

Regarding char signedness, how do I indicate it? `arch="x86_64* i686*"` ? I couldn't find a `XBPS_*` flag for it.

Reply:
you can't, they need to fix it

Reply:
`vsed`ing the version seems to have done the trick.

Do you know of a test I could run to check how those pkg-config flags are being propagated across the build system?

```
The following warnings were emitted during compilation:

warning: cc1: sorry, unimplemented: 64-bit mode not compiled in
warning: cc1: sorry, unimplemented: 64-bit mode not compiled in

error: failed to run custom build command for `libz-sys v1.0.25`

Caused by:
  process didn't exit successfully: `/builddir/ugdb-0.1.7/target/release/build/libz-sys-cf7ed1fe02cbb4c6/build-script-build` (exit code: 1)
--- stdout
cargo:rerun-if-env-changed=LIBZ_SYS_STATIC
cargo:rerun-if-changed=build.rs
cargo:rerun-if-env-changed=ZLIB_NO_PKG_CONFIG
cargo:rerun-if-env-changed=PKG_CONFIG
cargo:rerun-if-env-changed=ZLIB_STATIC
cargo:rerun-if-env-changed=ZLIB_DYNAMIC
cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-musl
cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_musl
cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
cargo:rerun-if-env-changed=PKG_CONFIG_PATH
cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-musl
cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_musl
cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-musl
cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_musl
cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
TARGET = Some("x86_64-unknown-linux-musl")
OPT_LEVEL = Some("3")
HOST = Some("x86_64-unknown-linux-musl")
CC_x86_64-unknown-linux-musl = None
CC_x86_64_unknown_linux_musl = None
HOST_CC = None
CC = Some("i686-linux-musl-gcc")
CFLAGS_x86_64-unknown-linux-musl = None
CFLAGS_x86_64_unknown_linux_musl = None
HOST_CFLAGS = None
CFLAGS = Some("-fstack-clash-protection -D_FORTIFY_SOURCE=2 -O2 -pipe -march=i686   -I/usr/i686-linux-musl/usr/include")
CRATE_CC_NO_DEFAULTS = None
DEBUG = Some("false")
CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
running: "i686-linux-musl-gcc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-fstack-clash-protection" "-D_FORTIFY_SOURCE=2" "-O2" "-pipe" "-march=i686" "-I/usr/i686-linux-musl/usr/include" "-I" "src/zlib" "-fvisibility=hidden" "-DSTDC" "-D_LARGEFILE64_SOURCE" "-D_POSIX_SOURCE" "-o" "/builddir/ugdb-0.1.7/target/release/build/libz-sys-4d4fdfddee6ecc41/out/build/src/zlib/adler32.o" "-c" "src/zlib/adler32.c"
cargo:warning=cc1: sorry, unimplemented: 64-bit mode not compiled in
exit code: 1
running: "i686-linux-musl-gcc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-fstack-clash-protection" "-D_FORTIFY_SOURCE=2" "-O2" "-pipe" "-march=i686" "-I/usr/i686-linux-musl/usr/include" "-I" "src/zlib" "-fvisibility=hidden" "-DSTDC" "-D_LARGEFILE64_SOURCE" "-D_POSIX_SOURCE" "-o" "/builddir/ugdb-0.1.7/target/release/build/libz-sys-4d4fdfddee6ecc41/out/build/src/zlib/compress.o" "-c" "src/zlib/compress.c"
cargo:warning=cc1: sorry, unimplemented: 64-bit mode not compiled in
exit code: 1

--- stderr


error occurred: Command "i686-linux-musl-gcc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-fstack-clash-protection" "-D_FORTIFY_SOURCE=2" "-O2" "-pipe" "-march=i686" "-I/usr/i686-linux-musl/usr/include" "-I" "src/zlib" "-fvisibility=hidden" "-DSTDC" "-D_LARGEFILE64_SOURCE" "-D_POSIX_SOURCE" "-o" "/builddir/ugdb-0.1.7/target/release/build/libz-sys-4d4fdfddee6ecc41/out/build/src/zlib/adler32.o" "-c" "src/zlib/adler32.c" with args "i686-linux-musl-gcc" did not execute successfully (status code exit code: 1).



warning: build failed, waiting for other jobs to finish...
error: build failed
```